README.md
## AngularJS 1.X slider directive with no external dependencies

Looking for an Angular version (> 1.X)? We got you covered: https://github.com/angular-slider/ngx-slider (Thanks [@piotrdz](https://github.com/piotrdz) :heart: )!

<hr />

Status:
![Maintenance](https://img.shields.io/maintenance/support-only/2017.svg?style=flat-square)
[![npm version](https://img.shields.io/npm/v/angularjs-slider.svg?style=flat-square)](https://www.npmjs.com/package/angularjs-slider)
[![npm downloads](https://img.shields.io/npm/dm/angularjs-slider.svg?style=flat-square)](http://npm-stat.com/charts.html?package=angularjs-slider&from=2015-01-01)
[![Build Status](https://img.shields.io/travis/angular-slider/angularjs-slider/master.svg?style=flat-square)](https://travis-ci.org/angular-slider/angularjs-slider)
[![codecov.io](https://img.shields.io/codecov/c/github/angular-slider/angularjs-slider.svg?style=flat-square)](https://codecov.io/github/angular-slider/angularjs-slider?branch=master)
![License](https://img.shields.io/badge/License-MIT-blue.svg?style=flat-square)

Links:
[![Join the chat at https://gitter.im/rzajac/angularjs-slider](https://img.shields.io/badge/GITTER-join%20chat-1dce73.svg?style=flat-square)](https://gitter.im/rzajac/angularjs-slider?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
[![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](http://makeapullrequest.com)

> I'm looking for a maintainer for this project. I have lost my [Open Source Stamina](https://kentcdodds.com/blog/open-source-stamina) for this project and I will probably not push any code to this project anymore (unless, I find some motivation later). I will try to merge pull requests if some are submitted, but only if they are really clean.

Slider directive implementation for AngularJS 1.X, without any dependencies: [http://angular-slider.github.io/angularjs-slider](http://angular-slider.github.io/angularjs-slider/index.html).

- Mobile friendly
- Fast
- Well documented
- Customizable
- Simple to use
- Keyboard support
- Compatibility with jQuery Lite, ie. with full jQuery ( Thanks Jusas! https://github.com/Jusas)
- Supports right to left

**Horizontal**

![image](https://cloud.githubusercontent.com/assets/2678610/11419158/d51cee88-9425-11e5-9d3f-3f7d97a31c6f.png)

**Vertical**

![image](https://cloud.githubusercontent.com/assets/2678610/11419099/7f4c0e76-9425-11e5-98c6-615412291df1.png)

**Custom style**
![image](https://cloud.githubusercontent.com/assets/2678610/21595997/df0bb468-d134-11e6-8272-008baffedccd.png)

## Examples

- **Simple example for single slider:** [http://jsfiddle.net/ValentinH/qjvxn4fc/](http://jsfiddle.net/ValentinH/qjvxn4fc/)
- **Simple example for double slider:** [http://jsfiddle.net/ValentinH/hnyL2axs/](http://jsfiddle.net/ValentinH/hnyL2axs/)
- **Various examples:** [http://angular-slider.github.io/angularjs-slider](http://angular-slider.github.io/angularjs-slider/index.html)
- **Same examples with live code:** https://jsfiddle.net/ValentinH/954eve2L/

## Reporting issues

Make sure the report is accompanied by a reproducible demo. The ideal demo is created by forking [our standard jsFiddle](http://jsfiddle.net/ValentinH/qjvxn4fc/), adding your own code and stripping it down to an absolute minimum needed to demonstrate the bug.

## Common issues

### My slider is not rendered correctly on load

If the slider's parent element is not visible during slider initialization, the slider can't know when its parent becomes visible.
For instance, when displaying a slider inside an element which visibility is toggled using ng-show, you need to send an event to force it to redraw when you set your ng-show to true.

Here's an example of `refreshSlider` method that you should call whenever the slider becomes visible.

```js
vm.refreshSlider = function() {
  $timeout(function() {
    $scope.$broadcast('rzSliderForceRender')
  })
}
```

if you get some flickering issues, you can try to replace to `$timeout` call by `$scope.$$postDigest` as suggested by @maknapp in [this issue](https://github.com/angular-slider/angularjs-slider/issues/79#issuecomment-219213647).

**ng-show-example**: http://jsfiddle.net/ValentinH/nzL6ax43/

**UI-Boostrap tabs example**: http://jsfiddle.net/ValentinH/bo23er5w/

### Decimal value can't be typed in an input field linked to the slider

By default, the slider value is always rounded to the nearest step. A side effect is that when a input field is linked to the slider in order to enable a user to directly type a value, the value is rounded when it doesn't match the step. Even worse, when using decimal values, when a user will type "0.", the `.` will directly be truncated since the value is rounded.

**Solution**: To avoid the value to be rounded, you need to use the `enforceStep: false` option. Thus, the value can be modified externally without taking care of the step. See [#298](https://github.com/angular-slider/angularjs-slider/issues/298).

## Installation

### NPM

```
npm i angularjs-slider
```

or

```
yarn add angularjs-slider
```

#### Typescript Support

Typescript definition files are provided with this project. To use them, be sure you have the angular type definition peer dependency installed.

```
npm i @types/angular
```

or

### Bower

```
$ bower install --save angularjs-slider
```

or

### CDNJS

Directly use (replace `X.X.X` by the version you want to use):

- `https://cdnjs.cloudflare.com/ajax/libs/angularjs-slider/X.X.X/rzslider.min.js`
- `https://cdnjs.cloudflare.com/ajax/libs/angularjs-slider/X.X.X/rzslider.min.css`

## Project integration

### Imports

```html
<link rel="stylesheet" type="text/css" href="/path/to/angularjs-slider/dist/rzslider.css"/>
<script src="/path/to/angularjs/angular.min.js"></script>
<script src="/path/to/angularjs-slider/dist/rzslider.min.js"></script>
```

### Module

```javascript
angular.module('yourApp', ['rzSlider'])
```

### Single slider

```javascript
// In your controller
$scope.priceSlider = 150
```

```html
<div>
    <rzslider rz-slider-model="priceSlider"></rzslider>
</div>
```

Above example would render a slider from 0 to 150. If you need floor and ceiling values use `rz-slider-options` attribute and provide an object with `floor`and `ceil`.

```html
<div>
    <rzslider
         rz-slider-model="slider.value"
         rz-slider-options="slider.options"></rzslider>
</div>
```

```js
$scope.slider = {
  value: 150,
  options: {
    floor: 0,
    ceil: 450,
  },
}
```

If you don't want to bother with an object set in your javascript file, you can pass an anonymous object literal to the slider options:

```html
<div>
    <rzslider
         rz-slider-model="value"
         rz-slider-options="{floor: 0, ceil: 450}"></rzslider>
</div>
```

```js
$scope.value = 150
```

### Range slider

```javascript
// In your controller
$scope.slider = {
  min: 100,
  max: 180,
  options: {
    floor: 0,
    ceil: 450,
  },
}
```

```html
<rzslider
    rz-slider-model="slider.min"
    rz-slider-high="slider.max"
    rz-slider-options="slider.options"></rzslider>
```

## Directive attributes

**rz-slider-model**

> Model for low value slider. If only _rz-slider-model_ is provided single slider will be rendered.

**rz-slider-high**

> Model for high value slider. Providing both _rz-slider-model_ and _rz-slider-high_ will render range slider.

**rz-slider-tpl-url**

> If you need to use a custom template, you can do so by providing a template URL to the `rz-slider-tpl-url` attribute. The default template is [this one](https://github.com/angular-slider/angularjs-slider/blob/master/src/rzSliderTpl.html).

The following variables are available in the template as scope variables.

- `floorLabel`: The value set to `floor` in `rz-slider-options`
- `ceilLabel`: The value set to `ceil` in `rz-slider-options`
- `modelLabel`: The value set to `rz-slider-model`
- `highLabel`: The value set to `rz-slider-high`
- `cmbLabel`: The text shown when the two handlers are close to each other. (e.g. "30-40")

The library replaces the HTML contents of label elements in the template by default, if you want to stop this behaviour and tweak label HTML on your own, you need to set `no-label-injection` class on the elements you're customizing.

See the [Custom template to use angular directive for label](./demo/directiveInCustomTemplate.html) for an example.

**rz-slider-options**

> An object with all the other options of the slider. Each option can be updated at runtime and the slider will automatically be re-rendered.

The default options are:

```js
{
    floor: 0,
    ceil: null, //defaults to rz-slider-model
    step: 1,
    precision: 0,
    minLimit: null,
    maxLimit: null,
    restrictedRange: null,
    minRange: null,
    maxRange: null,
    pushRange: false,
    id: null,
    translate: null,
    getLegend: null,
    stepsArray: null,
    bindIndexForStepsArray: false,
    draggableRange: false,
    draggableRangeOnly: false,
    showSelectionBar: false,
    showSelectionBarEnd: false,
    showOuterSelectionBars: false,
    showSelectionBarFromValue: null,
    hidePointerLabels: false,
    hideLimitLabels: false,
    autoHideLimitLabels: true,
    readOnly: false,
    disabled: false,
    interval: 350,
    showTicks: false,
    showTicksValues: false,
    ticksArray: null,
    ticksTooltip: null,
    ticksValuesTooltip: null,
    vertical: false,
    getSelectionBarColor: null,
    getTickColor: null,
    getPointerColor: null,
    keyboardSupport: true,
    scale: 1,
    enforceStep: true,
    enforceRange: false,
    noSwitching: false,
    onlyBindHandles: false,
    onStart: null,
    onChange: null,
    onEnd: null,
    rightToLeft: false,
    reversedControls: false,
    boundPointerLabels: true,
    mergeRangeLabelsIfSame: false,
    labelOverlapSeparator: ' - ',
    customTemplateScope: null,
    logScale: false,
    customValueToPosition: null,
    customPositionToValue: null,
    selectionBarGradient: null,
    ariaLabel: null,
    ariaLabelledBy: null,
    ariaLabelHigh: null,
    ariaLabelledByHigh: null,
    disableAnimation: false
}
```

**floor** - _Number (defaults to 0)_: Minimum value for a slider.

**ceil** - _Number (defaults to `rz-slider-model`value)_: Maximum value for a slider.

**step** - _Number (defaults to 1)_: Step between each value.

**precision** - _Number (defaults to 0)_: The precision to display values with (number of decimals to be displayed). The `toFixed()` is used internally for this.

**minLimit** - _Number (defaults to null)_: The minimum value authorized on the slider.

**maxLimit** - _Number (defaults to null)_: The maximum value authorized on the slider.

**restrictedRange** - _Object (defaults to null)_: Has two _Number_ properties, _from_ and _to_ that determine the bounds of an area that is not authorized for values. _Applies to range slider only._

**minRange** - _Number (defaults to null)_: The minimum range authorized on the slider. _Applies to range slider only._

**maxRange** - _Number (defaults to null)_: The maximum range authorized on the slider. _Applies to range slider only._

**pushRange** - _Boolean (defaults to false)_: Set to true to have a push behavior. When the min handle goes above the max, the max is moved as well (and vice-versa). The range between min and max is defined by the `step` option (defaults to 1) and can also be override by the `minRange` option. _Applies to range slider only._

**translate** - _Function(value, sliderId, label)_: Custom translate function. Use this if you want to translate values displayed on the slider.
`sliderId` can be used to determine the slider for which we are translating the value. `label` is a string that can take the following values:

- _'model'_: the model label
- _'high'_: the high label
- _'floor'_: the floor label
- _'ceil'_: the ceil label
- _'tick-value'_: the ticks labels

For example if you want to display dollar amounts instead of just numbers:

```html
<div>
    <rzslider
         rz-slider-model="slider.value"
         rz-slider-options="slider.options"></rzslider>
</div>
```

```js
$scope.slider = {
  value: 0,
  options: {
    floor: 0,
    ceil: 100,
    translate: function(value) {
      return '$' + value
    },
  },
}
```

**getLegend** - _Function(value, sliderId)_: Use to display legend under ticks (thus, it needs to be used along with `showTicks` or `showTicksValues`). The function will be called with each tick value and returned content will be displayed under the tick as a legend. If the returned value is null, then no legend is displayed under the corresponding tick.You can also directly provide the legend values in the `stepsArray` option.

> In order to get enough space to display legends under the slider, you need to add the `with-legend` class to the slider component. The default margin-bottom is then 40px which is enough for legends that are displayed on 2 lines. If you need more, simply override the style for the class.

**id** - _Any (defaults to null)_: If you want to use the same `translate` function for several sliders, just set the `id` to anything you want, and it will be passed to the `translate(value, sliderId)` function as a second argument.

**stepsArray** - _Array_: If you want to display a slider with non linear/number steps.
Just pass an array with each slider value and that's it; the floor, ceil and step settings of the slider will be computed automatically. By default, the `rz-slider-model` and `rz-slider-high` values will be the value of the selected item in the stepsArray. They can also be bound to the index of the selected item by setting the `bindIndexForStepsArray` option to `true`.

`stepsArray` can also be an array of objects or Dates like:

```js
;[
  { value: 'A' }, // the display value will be *A*
  { value: 10, legend: 'Legend for 10' }, // the display value will be 10 and a legend will be displayed under the corresponding tick.
  new Date(2016, 7, 12), // the display value will be the default format of Date. To customize it, use the `translate` option
  { value: new Date(2016, 7, 12), legend: 'Legend for 10' }, // same as above but with a legend
]
```

**bindIndexForStepsArray** - _Boolean (defaults to false)_: Set to true to bind the index of the selected item to `rz-slider-model` and `rz-slider-high`. (This was the default behavior prior to 4.0).


CHANGELOG.md
# 7.0.1 (2021-09-06)

## Bug fix

- chore(styles): add parentheses around division-like expr's (#679)

# 7.0.0 (2019-02-23)

## Feature

- feat(transition): slider moves are now animated. (this can be disabled via the `disableAnimation` option)

## Bug fix

- fix(vertical): Support switching between horizontal and vertical

## Breaking change

The module name was renamed from `rzModule` to `rzSlider`.

# 6.7.0 (2019-02-23)

## Feature

- feat(ticks): add the option to use ticksArray as array of objects (#662, thanks @vdiez)

# 6.6.0 (2018-06-29)

## Feature

- feat(range): adds the `restrictedRange` option (#638, thanks @DanielReid)

## ⚠️ Unwanted breaking change ⚠️

Sorry for this, but this version is breaking if you are using the `rz-slider-tpl-url` attribute. You need to update your template to have the correct number of elements.

# 6.5.1 (2018-03-30)

## Bug fix

- revert: onStart: Remove stop propagation call so events will bubble up (#612)

# 6.5.0 (2018-02-06)

## Feature

- feat(\$compile): adds labelOverlapSeparator as an option (#616)

# 6.4.4 (2018-01-24)

## Bug fix

- onStart: Remove stop propagation call so events will bubble up (#612)

# 6.4.3 (2017-12-01)

## Bug fix

- Remove the semicolon at the end of the `dist/rzslider.css` file. It was introduced by previous release.

# 6.4.2 (2017-11-30)

## Bug fix

- Add a semicolon at the end of the `dist/rzslider.js` file. It avoids errors when people concat this file with other libs without using the minified version.

# 6.4.1 (2017-11-17)

## Bug fix

- Options: react to changes of options of Function type (#590)

# 6.4.0 (2017-08-12)

## Feature

- Add the `showOuterSelectionBars` option (#553).

# 6.3.0 (2017-08-07)

## Feature

- Handle different values for `showTicks` and `showTicksValues` (#550).

# 6.2.3 (2017-07-08)

## Tooling

- Add Typescript definition file.

# 6.2.2 (2017-05-16)

## Fixes

- Fix (again) onEnd event de-registration.

# 6.2.1 (2017-05-15)

## Fixes

- Fix onEnd event being sent several times on non-mobiles devices (#536)

# 6.2.0 (2017-05-25)

## New Feature

- Handle multi touch events on separate sliders (#535). Thanks @daniela-mateescu :)

# 6.1.2 (2017-05-15)

## Fixes

- Fix ticks and values at intermediate positions on IE (#531)

# 6.1.1 (2017-03-29)

## Fixes

- Add vendor prefixes for transform property in JS code (#518)

# 6.1.0 (2017-03-06)

## Features

- Add labelling options for a11y (#505)

# 6.0.2 (2017-03-02)

## Fixes

- Update the combined labels on separation (#502)

# 6.0.1 (2017-02-14)

## Fixes

- Ensure model value is current when custom translate function runs for tick values

# 6.0.0 (2017-01-02)

## Refactoring

- Refactor/simplify the css rules to ease the customisation.

**You might want to check that all your custom styles are still correctly applied...**

# 5.9.0 (2016-12-12)

## Features

- Add selectionBarGradient option to customize the selection bar (#473)

# 5.8.9 (2016-12-11)

## Improvement

- Add autoprefixer for CSS builds (#472)

# 5.8.8 (2016-12-11)

## Fix

- Prevent angular being loaded twice when using with browserify (#474)

# 5.8.7 (2016-11-09)

## Fix

- Add Math.round for positions and dimensions - thanks to @DmitryKrekota (#454)

# 5.8.6 (2016-11-08)

## Fix

- Apply the pushRange with maxRange - thanks to @GuilloOme (#456)

# 5.8.5 (2016-11-05)

## Fix

- Fix overlapping max and ceil labels in some cases (#396)

# 5.8.4 (2016-11-05)

## Improvement

- Refactor autoHiding algorithm for labels (fix #446)

# 5.8.3 (2016-11-03)

## Improvement

- Generate a SCSS file (simple copy of the css file) in the dist folder so it can be imported (#449)

# 5.8.2 (2016-11-03)

## Fix

- Fix ceil label positioning (#448)

# 5.8.1 (2016-10-27)

## Fix

- Enable using with Browserify (#436)

# 5.8.0 (2016-10-22)

## Features

- Handle Date object in stepsArray (#424 )

## Fixes

- Fix style for disabled range slider and ticks (#394)
- Fix slider goes back when moved and scaled (#346)

# 5.7.0 (2016-10-16)

## Features

- Add a `logScale` option to display the slider using a logarithmic scale (#280).
- Add `customValueToPosition` and `customPositionToValue` options to display the slider using a custom scale (#280).

# 5.6.0 (2016-10-16)

## Features

- Add a `ticksArray` option to display ticks at specific positions (#426).

To enable this new feature, the way the ticks are rendered has been changed. Now each tick is positioned absolutely using a `transform: translate()` instruction.

# 5.5.1 (2016-09-22)

## Fix

- Prevent losing focus when slider is rerendered (#415).

# 5.5.0 (2016-09-06)

## Features

- Add an `autoHideLimitLabels` to disable the auto-hiding of limit labels (#405).

# 5.4.3 (2016-08-07)

## Fix

- Fix minLimit/maxLimit bugged for draggableRange (#384).

# 5.4.2 (2016-08-02)

## Fix

- Fix minimum value goes below floor when using maxRange (#377).

# 5.4.1 (2016-07-17)

## Fix

- Fix showing limit labels when pointer labels are always hidden (#373).

# 5.4.0 (2016-07-13)

## Features

- Add function to customize color of ticks (#372).

# 5.3.0 (2016-07-11)

## Features

- Expose labels on scope in template (#358).

# 5.2.0 (2016-07-07)

## Features

- Add a `customTemplateScope` option (#354).

# 5.1.1 (2016-07-06)

## Fix

- Fix the way to check when event properties are undefined (#365).

# 5.1.0 (2016-07-02)

## Features

- Add a `pushRange` option (#341).

# 5.0.1 (2016-07-01)

## Fix

- Switch from using opacity to visibility to show/hide elements (#362).

# 5.0.0 (2016-06-30)

## Fix

- AMD/CommonJS exported module: export module name instead of module (#360).

## Breaking change

Code that relies on the module object to be exported (accessing the name via .name for example) will break, since the name is now directly returned.

# 4.1.0 (2016-06-30)

## Improvement

- Add a `bindIndexForStepsArray` option that enable to use `stepsArray` with the same behavior as before 4.0 (#345).

## Fix


rzslider.d.ts
/**
 * Typescript type definition file.
 */

import * as angular from "angular";

declare module "angular" {
    export namespace RzSlider {
        type RzLabelType = "model"|"high"|"floor"|"ceil"|"tick-value";
        type RzPointerType = "min"|"max";
        type RzCallback = (id: string, modelValue: number, highValue: number, pointerType: "min"|"max") => void;
        type RzTranslate = (value: number, sliderId: string, label: RzLabelType) => string;

        /** RZ slider options typing */
        interface RzOptions {
            /** Number (defaults to 0): Minimum value for a slider. */
            floor?: number;
            /** Number (defaults to rz-slider-modelvalue): Maximum value for a slider. */
            ceil?: number;
            /** Number (defaults to 1): Step between each value. */
            step?: number;
            /** Number (defaults to 0): The precision to display values with. The toFixed() is used internally for this. */
            precision?: number;
            /** Number (defaults to null): The minimum value authorized on the slider. */
            minLimit?: number;
            /** Number (defaults to null): The maximum value authorized on the slider. */
            maxLimit?: number;
            /** Number (defaults to null): The minimum range authorized on the slider. Applies to range slider only. */
            minRange?: number;
            /** Number (defaults to null): The maximum range authorized on the slider. Applies to range slider only. */
            maxRange?: number;
            /**
             * Boolean (defaults to false): Set to true to have a push behavior. When the min handle goes above the max,
             * the max is moved as well (and vice-versa). The range between min and max is defined by the step option
             * (defaults to 1) and can also be override by the minRange option. Applies to range slider only.
             */
            pushRange?: boolean;
            /**
             * Custom translate function. Use this if you want to translate values displayed on the slider.
             * sliderId can be used to determine the slider for which we are translating the value.
             * label is a string that can take the following values:
             * 'model': the model label
             * 'high': the high label
             * 'floor': the floor label
             * 'ceil': the ceil label
             * 'tick-value': the ticks labels
             */
            translate?: RzTranslate;
            /**
             * Function(value, sliderId): Use to display legend under ticks. The function will be called with each tick
             * value and returned content will be displayed under the tick as a legend. If the returned value is null,
             * then no legend is displayed under the corresponding tick.
             */
            getLegend?: (value: number, sliderId: string) => string;
            /**
             * Any (defaults to null): If you want to use the same translate function for several sliders,
             * just set the id to anything you want, and it will be passed to the translate(value, sliderId)
             * function as a second argument.
             */
            id?: string;
            /**
             * Array: If you want to display a slider with non linear/number steps. Just pass an array with each slider
             * value and that's it; the floor, ceil and step settings of the slider will be computed automatically.
             * By default, the rz-slider-model and rz-slider-high values will be the value of the selected item in the stepsArray.
             * They can also be bound to the index of the selected item by setting the bindIndexForStepsArray option to true.
             *
             * stepsArray can also be an array of objects like:
             * [
             * {value: 'A'}, // the display value will be *A*
             * {value: 10, legend: 'Legend for 10'} // the display value will be 10 and a legend will be displayed under the corresponding tick.
             * ]
             */
            stepsArray?: any[];
            /**
             * Boolean (defaults to false): Set to true to bind the index of the selected item to rz-slider-model and rz-slider-high.
             * (This was the default behavior prior to 4.0).
             */
            bindIndexForStepsArray?: boolean;
            /** Boolean (defaults to false): When set to true and using a range slider, the range can be dragged by the selection bar. Applies to range slider only. */
            draggableRange?: boolean;
            /** Boolean (defaults to false): Same as draggableRange but the slider range can't be changed. Applies to range slider only. */
            draggableRangeOnly?: boolean;
            /** Boolean (defaults to false): Set to true to always show the selection bar before the slider handle. */
            showSelectionBar?: boolean;
            /** Boolean (defaults to false): Set to true to always show the selection bar after the slider handle. */
            showSelectionBarEnd?: boolean;
            /** Boolean (defaults to false): Only for range slider. Set to true to visualize in different colour the areas on the left/right (top/bottom for vertical range slider) of selection bar between the handles. */
            showOuterSelectionBars?: boolean;
            /** Number (defaults to null): Set a number to draw the selection bar between this value and the slider handle. */
            showSelectionBarFromValue?: number;
            /**
             * Function(value) or Function(minVal, maxVal) (defaults to null): Function that returns the current color of the
             * selection bar. If your color won't changed, don't use this option but set it through CSS. If the returned color
             * depends on a model value (either rzScopeModelor 'rzSliderHigh), you should use the argument passed to the function.
             * Indeed, when the function is called, there is no certainty that the model has already been updated.
             */
            getSelectionBarColor?: (minVal: number, maxVal?: number) => string;
            /** Function(value) (defaults to null): Function that returns the color of a tick. showTicks must be enabled. */
            getTickColor?: (value: number) => string;
            /**
             * Function(value, pointerType) (defaults to null): Function that returns the current color of a pointer.
             * If your color won't changed, don't use this option but set it through CSS. If the returned color depends
             * on a model value (either rzScopeModelor 'rzSliderHigh), you should use the argument passed to the function.
             * Indeed, when the function is called, there is no certainty that the model has already been updated.
             * To handle range slider pointers independently, you should evaluate pointerType within the given function
             * where "min" stands for rzScopeModel and "max" for rzScopeHigh values.
             */
            getPointerColor?: (value: number, pointerType: RzPointerType) => string;
            /** Boolean (defaults to false): Set to true to hide pointer labels */
            hidePointerLabels?: boolean;
            /** Boolean (defaults to false): Set to true to hide min / max labels */
            hideLimitLabels?: boolean;
            /** Boolean (defaults to true): Set to false to disable the auto- hiding behavior of the limit labels. */
            autoHideLimitLabels?: boolean;
            /** Boolean (defaults to false): Set to true to make the slider read-only. */
            readOnly?: boolean;
            /** Boolean (defaults to false): Set to true to disable the slider. */
            disabled?: boolean;
            /**
             * Number in ms (defaults to 350): Internally, a throttle function (See http://underscorejs.org/#throttle) is used
             * when the model or high values of the slider are changed from outside the slider. This is to prevent from re-rendering
             * the slider too many times in a row. interval is the number of milliseconds to wait between two updates of the slider.
             */
            interval?: number;
            /** Boolean or Number (defaults to false): Set to true to display a tick for each step of the slider. Set an integer to display ticks at intermediate positions. */
            showTicks?: boolean | number;
            /** Boolean or Number (defaults to false): Set to true to display a tick and the step value for each step of the slider. Set an integer to display ticks and the step value at intermediate positions. */
            showTicksValues?: boolean | number;
            /** Array (defaults to null): Use to display ticks at specific positions. The array contains the index of the ticks that should be displayed. For example, [0, 1, 5] will display a tick for the first, second and sixth values. */
            ticksArray?: number[];
            /** Function(value) (defaults to null): (requires angular-ui bootstrap) Used to display a tooltip when a tick is hovered. Set to a function that returns the tooltip content for a given value. */
            ticksTooltip?: (value: number) => string;
            /** Function(value) (defaults to null): Same as ticksTooltip but for ticks values. */
            ticksValuesTooltip?: (value: number) => string;
            /** Number (defaults to 1): If you display the slider in an element that uses transform: scale(0.5), set the scale value to 2 so that the slider is rendered properly and the events are handled correctly. */
            scale?: number;
            /** Boolean (defaults to true): Set to true to force the value to be rounded to the step, even when modified from the outside.. When set to false, if the model values are modified from outside the slider, they are not rounded and can be between two steps. */
            enforceStep?: boolean;
            /** Boolean (defaults to false): Set to true to round the rzSliderModel and rzSliderHigh to the slider range even when modified from outside the slider. When set to false, if the model values are modified from outside the slider, they are not rounded but they are still rendered properly on the slider. */
            enforceRange?: boolean;
            /** Boolean (defaults to false): Set to true to prevent to user from switching the min and max handles. Applies to range slider only. */
            noSwitching?: boolean;
            /** Boolean (defaults to false): Set to true to only bind events on slider handles. */
            onlyBindHandles?: boolean;
            /** Boolean (defaults to true): Set to true to keep the slider labels inside the slider bounds. */
            boundPointerLabels?: boolean;
            /** Boolean (defaults to false): Set to true to merge the range labels if they are the same. For instance, if min and max are 50, the label will be "50 - 50" if mergeRangeLabelsIfSame: false, else "50". */
            mergeRangeLabelsIfSame?: boolean;
            /** String (defaults to ' - '): Separator to use when the labels overlap. For instance, if min and max are -1 and 1, the label will be "-1 .. 1" if `labelOverlapSeparator: ' .. '`. */
            labelOverlapSeparator?: string;
            /** Function(sliderId, modelValue, highValue, pointerType): Function to be called when a slider update is started. If an id was set in the options, then it's passed to this callback. This callback is called before any update on the model. pointerType is either 'min' or 'max' depending on which handle is used. */
            onStart?: RzCallback;
            /**
             * Function to be called when rz-slider-model or rz-slider-high change. If an id was set in the options,
             * then it's passed to this callback. pointerType is either 'min' or 'max' depending
             * on which handle is used.
             */
            onChange?: RzCallback;
            /** Function(sliderId, modelValue, highValue, pointerType): Function to be called when a slider update is ended. If an id was set in the options, then it's passed to this callback. pointerType is either 'min' or 'max' depending on which handle is used. */
            onEnd?: RzCallback;
            /** Boolean (defaults to false): Set to true to show graphs right to left. If vertical is true it will be from top to bottom and left / right arrow functions reversed. */
            rightToLeft?: boolean;
            /**
             * Boolean (defaults to false): Set to true to display the slider vertically. The slider will take the full height of its parent.
             * Changing this value at runtime is not currently supported.
             */
            vertical?: boolean;
            /**
             * Boolean (defaults to true): Handles are focusable (on click or with tab) and can be modified using the following keyboard controls:
             *   - Left/bottom arrows: -1
             *   - Right/top arrows: +1
             *   - Page-down: -10%
             *   - Page-up: +10%
             *   - Home: minimum value
             *   - End: maximum value
             */
            keyboardSupport?: boolean;
            /**
             * Boolean (defaults to false): Set to true to reverse keyboard navigation:
             *  - Right/top arrows: -1
             *  - Left/bottom arrows: +1
             *  - Page-up: -10%
             *  - Page-down: +10%
             *  - End: minimum value
             *  - Home: maximum value
             */
            reversedControls?: boolean;
            /** Object (default to null): The properties defined in this object will be exposed in the slider template under custom.X. */
            customTemplateScope?: any;
            /** Boolean (defaults to false): Set to true to use a logarithmic scale to display the slider. */
            logScale?: boolean;
            /** Function(val, minVal, maxVal): percent: Function that returns the position on the slider for a given value.The position must be a percentage between 0 and 1. */
            customValueToPosition?: (val: number, minVal: number, maxVal: number) => number;
            /** Function(percent, minVal, maxVal): value: Function that returns the value for a given position on the slider. The position is a percentage between 0 and 1. */
            customPositionToValue?: (percent: number, minVal: number, maxVal: number) => number;
            /** Object(default to null): Use to display the selection bar as a gradient. The given object must contain from and to properties which are colors. */
            selectionBarGradient?: {from: string, to: string};
            /** String(default to null): Use to add a label directly to the slider(s) for accessibility. Adds the aria-label attribute. */
            ariaLabel?: string;
            /** String(default to null): Use to add a label directly to the slider(s) for accessibility. Adds the aria-label attribute. */
            ariaLabelHigh?: string;
            /** String(default to null): Use instead of ariaLabel and ariaLabelHigh to reference the id of an element which will be used to label the slider(s). Adds the aria-labelledby attribute. */
            ariaLabelledBy?: string;
            /** String(default to null): Use instead of ariaLabel and ariaLabelHigh to reference the id of an element which will be used to label the slider(s). Adds the aria-labelledby attribute. */
            ariaLabelledByHigh?: string;
        }
    }
}
